lib/galaxy/tools/parameters/basic.py

```diff
@@ -676,36 +676,19 @@ def get_initial_value( self, trans, context, history=None ):
 ## it for now to avoid breaking any tools.
 
 
-class BaseURLToolParameter( ToolParameter ):
+class BaseURLToolParameter( HiddenToolParameter ):
     """
     Returns a parameter the contains its value prepended by the
     current server base url. Used in all redirects.
     """
-
-    def __init__( self, tool, input_source ):
-        input_source = ensure_input_source( input_source )
-        ToolParameter.__init__( self, tool, input_source )
-        self.value = input_source.get( 'value', '' )
-
-    def get_value( self, trans ):
-        # url = trans.request.base + self.value
-        url = url_for( self.value, qualified=True )
-        return url
-
-    def get_html_field( self, trans=None, value=None, other_values={} ):
-        return form_builder.HiddenField( self.name, self.get_value( trans ) )
-
     def get_initial_value( self, trans, context, history=None ):
-        return self.value
+        self.url_for = url_for( '', qualified=True )
```

Comment:
Am I reading this correctly - you are setting a persistent object instance variable on every  web request? I guess so that ``to_param_dict_string`` has a value? This would fail however if ``to_param_dict_string`` was ever called on say a job handler or in a web thread before a request has been made - have you verified these are not problems by running such jobs on a multi-process server with separate job threads?

I imagine this will cause really confused behavior if Galaxy is being served on multiple hosts - e.g. if an admin loads the app directly on localhost for instance - a seemingly common scenario. I think the right approach here is to always generate this value on the client side if at all possible?